File: Data Structures & Algorithms/add-two-numbers/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode l3 = new ListNode();
        ListNode curr = l3;
       
        int carry =0;
        while(l1 != null || l2 !=null || carry !=0){
            int v1 = (l1 != null) ? l1.val : 0;
            int v2 = (l2 != null) ? l2.val : 0;
           int sum = v1 + v2 + carry;
            carry = sum / 10;
            sum = sum % 10;
            curr.next = new ListNode(sum);

            curr = curr.next;
            l1 = (l1 != null) ? l1.next : null;
            l2 = (l2 != null) ? l2.next : null;
        }

        return l3.next;
    }
}

File: Data Structures & Algorithms/reorder-linked-list/submission-0.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {void}
     */
    reorderList(head: ListNode | null): void {

            let s = head;
            let f = head;

            while(f != null && f.next != null){
                s = s.next;
                f = f.next!.next;                
            }
            
            let middle = s;

            let prev : ListNode | null = null;
            let curr = middle;

            while(curr != null){
                let next = curr.next;
                curr.next = prev;
                prev = curr;
                curr = next;
            }

            middle = prev;

            let first = head;
            let second = middle;

            while(second.next !== null){
                let fnext = first!.next;
                let snext = second!.next;

                first.next = second;
                second.next = fnext;

                first = fnext;
                second = snext; 
            }


    }
}
